libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.java

```diff
@@ -105,69 +95,42 @@ public class PushService extends JobIntentService {
-		 */
-		final Bundle bundle = intent.getBundleExtra(PushMessaging.ACCOUNT_BUNDLE_KEY);
-		UserAccount account = null;
-		boolean allAccounts = false;
-		if (bundle != null) {
-			final String allAccountsValue = bundle.getString(PushMessaging.ACCOUNT_BUNDLE_KEY);
-			if (PushMessaging.ALL_ACCOUNTS_BUNDLE_VALUE.equals(allAccountsValue)) {
-				allAccounts = true;
-			} else {
-				account = new UserAccount(bundle);
-			}
-		}
-		final UserAccountManager userAccMgr = SalesforceSDKManager.getInstance().getUserAccountManager();
-		final List<UserAccount> accounts = userAccMgr.getAuthenticatedUsers();
-		boolean register = SFDC_REGISTRATION_RETRY_INTENT.equals(intent.getAction());
-		boolean unregister = SFDC_UNREGISTRATION_INTENT.equals(intent.getAction());
-		if (register || unregister) {
-			if (allAccounts) {
-				if (accounts != null) {
-					for (final UserAccount userAcc : accounts) {
-
-						/*
-						 * If 'register' is true, we are registering, if it's false, we must be
-						 * un-registering, because of the if gate above.
-						 */
-						performRegistrationChange(register, userAcc);
-					}
-				}
-			} else {
-				if (account == null) {
-					account = userAccMgr.getCurrentUser();
-				}
-
-				/*
-				 * If 'register' is true, we are registering, if it's false, we must be
-				 * un-registering, because of the if gate above.
-				 */
-				performRegistrationChange(register, account);
-			}
-		}
+	/**
+	 * Enqueues a change to one or more user accounts' push notifications
+	 * registration as persistent work via Android background tasks.
+	 *
+	 * @param userAccount       The user account or null for all user accounts
+	 * @param action            The push notifications registration action
+	 * @param delayMilliseconds The amount of delay before the push registration
+	 *                          action is taken, such as in a retry scenario
+	 */
+	static void enqueuePushNotificationsRegistrationWork(
+			@Nullable UserAccount userAccount,
+			@NonNull PushNotificationsRegistrationAction action,
+			@Nullable Long delayMilliseconds) {
+		final Context context = SalesforceSDKManager.getInstance().getAppContext();
+		final WorkManager workManager = WorkManager.getInstance(context);
+		final String userAccountJson = userAccount == null ? null : userAccount.toJson().toString();
+		final Data workData = new Data.Builder().putString(
+				"USER_ACCOUNT",
+				userAccountJson
+		).putString(
+				"ACTION",
+				action.name()
+		).build();
+		final OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(
+				PushNotificationsRegistrationChangeWorker.class
+		).setInputData(
+				workData
+		).setInitialDelay(
+				delayMilliseconds == null ? 0 : delayMilliseconds,
+				MILLISECONDS
+		).build();
+
+		workManager.enqueue(workRequest);
 	}
 
-	private void performRegistrationChange(boolean register, UserAccount userAccount) {
+	void performRegistrationChange(boolean register, UserAccount userAccount) {
```

Comment:
This moves to package level access to be invoked by `PushNotificationsRegistrationChangeWorker`.